Non-descriptive error message if webpack.config.js is invalid

Take the Spring Boot starter project and replace `webpack.config.js` with 
```
foo
```

If you run `node_modules/.bin/webpack` manually, you will get
```
ReferenceError: foo is not defined
    at Object.<anonymous> (.../webpack.config.js:1:63)
```

If you run `mvn`, you get
```
2020-02-03 12:15:47.107 ERROR 60528 --- [  restartedMain] o.a.c.c.C.[Tomcat].[localhost].[/]       : Exception sending context initialized event to listener instance of class [com.vaadin.flow.spring.VaadinServletContextInitializer$DevModeServletContextListener]

java.lang.RuntimeException: Unable to initialize Vaadin DevModeHandler
	at com.vaadin.flow.spring.VaadinServletContextInitializer$DevModeServletContextListener.contextInitialized(VaadinServletContextInitializer.java:349) ~[vaadin-spring-12.2.0.alpha1.jar:na]
	at org.apache.catalina.core.StandardContext.listenerStart(StandardContext.java:4680) [tomcat-embed-core-9.0.22.jar:9.0.22]
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5143) [tomcat-embed-core-9.0.22.jar:9.0.22]
...
Caused by: java.lang.IllegalStateException: Webpack exited prematurely
	at com.vaadin.flow.server.DevModeHandler.<init>(DevModeHandler.java:201) ~[flow-server-2.2.0.alpha10.jar:2.2.0.alpha10]
	at com.vaadin.flow.server.DevModeHandler.createInstance(DevModeHandler.java:298) ~[flow-server-2.2.0.alpha10.jar:2.2.0.alpha10]
	at com.vaadin.flow.server.DevModeHandler.start(DevModeHandler.java:248) ~[flow-server-2.2.0.alpha10.jar:2.2.0.alpha10]

```

This does not even contain a reference to `webpack.config.js` nor to the problem that actually occured in `webpack`

Same thing here with Vaadin 14.5.1: I only see the exception but there is no indication in the entire stdout as to why the webpack would fail:
```
Caused by: java.lang.IllegalStateException: Webpack exited prematurely
        at com.vaadin.flow.server.DevModeHandler.doStartWebpack(DevModeHandler.java:657)
        at com.vaadin.flow.server.DevModeHandler.doStartDevModeServer(DevModeHandler.java:583)
        at com.vaadin.flow.server.DevModeHandler.runOnFutureComplete(DevModeHandler.java:530)
        at com.vaadin.flow.server.DevModeHandler.lambda$new$0(DevModeHandler.java:146)
        at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:774)
```
This started to happen after I applied the checker from #8475 . Without the checker the Vaadin would fail in runtime with `lumo-styles already registered`: the checker apparently catches the issue early on (which is a good thing), but then Vaadin logs nothing of the issue and fails with this useless exception.

--- a/flow-server/src/main/java/com/vaadin/flow/server/DevModeHandler.java
+++ b/flow-server/src/main/java/com/vaadin/flow/server/DevModeHandler.java
@@ -109,6 +109,8 @@ public final class DevModeHandler implements RequestHandler {
 
     private boolean notified = false;
 
+    private StringBuilder cumulativeOutput = new StringBuilder();
+
     private volatile String failedOutput;
 
     private AtomicBoolean isDevServerFailedToStart = new AtomicBoolean();
@@ -469,6 +471,9 @@ public final class DevModeHandler implements RequestHandler {
                 output.append(cleanLine).append(System.lineSeparator());
             }
 
+            // save output so as it can be used to log exception if run fails
+            cumulativeOutput.append(cleanLine).append(System.lineSeparator());
+
             boolean succeed = success.matcher(line).find();
             boolean failed = failure.matcher(line).find();
             // We found the success or failure pattern in stream
@@ -478,6 +483,7 @@ public final class DevModeHandler implements RequestHandler {
                 failedOutput = failed ? output.toString() : null;
                 // reset output and logger for the next compilation
                 output = getOutputBuilder();
+                cumulativeOutput = new StringBuilder();
                 log = info;
                 // Notify DevModeHandler to continue
                 doNotify();
@@ -654,6 +660,9 @@ public final class DevModeHandler implements RequestHandler {
             }
 
             if (!webpackProcess.get().isAlive()) {
+                getLogger().error(
+                    String.format("Webpack failed with the exception:%n%s",
+                        cumulativeOutput.toString()));
                 throw new IllegalStateException("Webpack exited prematurely");
             }
 

